crates/forge/tests/integration/test_state.rs

```diff
@@ -0,0 +1,134 @@
+
+use crate::integration::common::runner::Contract;
+use crate::integration::common::running_tests::run_test_case;
+use crate::{assert_passed, test_case};
+use indoc::indoc;
+
+#[test]
+fn test_storage_access_from_tests() {
+    let test = test_case!(indoc!(
+        r#"
+        #[starknet::contract]
+        mod Contract {
+            #[storage]
+            struct Storage {
+                balance: felt252, 
+            }
+            
+            #[generate_trait]
+            impl InternalImpl of InternalTrait {
+                fn internal_function(self: @ContractState) -> felt252 {
+                    self.balance.read()
+                }
+            }
+        }
+
+        use ___PREFIX_FOR_PACKAGE___test_case::Contract::balance::InternalContractMemberStateTrait;
+
+        #[test]
+        fn test_internal() {
+            let mut state = Contract::contract_state_for_testing();
+            state.balance.write(10);
+            
+            let value = Contract::InternalImpl::internal_function(@state);
+            assert(value == 10, 'Incorrect storage value');
+        }
+    "#
+    ),);
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+fn test_simple_syscalls() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use starknet::info::{get_execution_info};
+        use result::ResultTrait;
+        use box::BoxTrait;
+        use starknet::info::TxInfo;
+        use serde::Serde;
+        use starknet::ContractAddress;
+        use array::SpanTrait;
+        use snforge_std::{ declare, ContractClassTrait };
+
+        #[starknet::interface]
+        trait ISpoofChecker<TContractState> {
+            fn get_tx_hash(ref self: TContractState) -> felt252;
+            fn get_nonce(ref self: TContractState) -> felt252;
+            fn get_account_contract_address(ref self: TContractState) -> ContractAddress;
+            fn get_signature(ref self: TContractState) -> Span<felt252>;
+            fn get_version(ref self: TContractState) -> felt252;
+            fn get_max_fee(ref self: TContractState) -> u128;
+            fn get_chain_id(ref self: TContractState) -> felt252;
+        }
+        #[starknet::interface]
+        trait IRollChecker<TContractState> {
+            fn get_block_number(ref self: TContractState) -> u64;
+        }
+
+        #[starknet::interface]
+        trait IWarpChecker<TContractState> {
+            fn get_block_timestamp(ref self: TContractState) -> u64;
+        }
+
+        #[test]
+        fn test_get_execution_info() {
+            let exec_info = get_execution_info().unbox();
+            assert(exec_info.caller_address.into() == 0, 'Incorrect caller address');
+            assert(exec_info.contract_address.into() == 0, 'Incorrect contract address');
+            assert(exec_info.entry_point_selector.into() == 1515033776670125170357353230573218612959527887889480774359330962438924478531, 'Incorrect entry point selector');
```

Comment:
Why this value?

Reply:
I think a comment explaining from what selector it is computed wouldn't hurt.

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/589/commits/4b8a9fc2b74b94ff7374c760376beb3152b4fc4d